diff --git a/src/frontends/tensorflow/src/frontend.cpp b/src/frontends/tensorflow/src/frontend.cpp
index 06944d27e..752572cac 100644
--- a/src/frontends/tensorflow/src/frontend.cpp
+++ b/src/frontends/tensorflow/src/frontend.cpp
@@ -183,7 +183,7 @@ void FrontEnd::translate_graph(const ov::frontend::InputModel::Ptr& model,
                 results.push_back(result);
             } else {
                 auto param = std::dynamic_pointer_cast<ov::opset8::Parameter>(output.get_node_shared_ptr());
-                if (param && operation_decoder->get_op_type() != "Identity") {
+                if (param && operation_decoder->get_op_type() != "Identity" & operation_decoder->get_op_type() != "FusedBatchNormV3") {
                     params.push_back(param);
                 }
                 ng_op_map[operation_name].push_back(output);
