diff --git a/official/transformer/translate.py b/official/transformer/translate.py
index 9e753ed9..00f56c0e 100644
--- a/official/transformer/translate.py
+++ b/official/transformer/translate.py
@@ -24,6 +24,7 @@ import os
 from absl import app as absl_app
 from absl import flags
 import tensorflow as tf
+import ngraph_bridge
 # pylint: enable=g-bad-import-order
 
 from official.transformer.utils import tokenizer
