diff --git a/BUILD.ovtf b/BUILD.ovtf
new file mode 100644
index 00000000..8def747a
--- /dev/null
+++ b/BUILD.ovtf
@@ -0,0 +1,5 @@
+cc_library(
+        name = "openvino_tensorflow",
+        srcs = ["libopenvino_tensorflow.so"],
+        visibility = ["//visibility:public"]
+)
diff --git a/WORKSPACE b/WORKSPACE
index 18af812a..faf81201 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -52,3 +53,10 @@ load("@rules_proto//proto:repositories.bzl", "rules_proto_dependencies", "rules_
 rules_proto_dependencies()
 rules_proto_toolchains()

+# External dependency to openvino tensorflow
+new_local_repository(
+        name="openvino_tensorflow",
+        path="/usr/local/lib/openvino_tensorflow",
+        build_file="BUILD.ovtf"
+)
+
diff --git a/tensorflow_serving/model_servers/BUILD b/tensorflow_serving/model_servers/BUILD
index 25236357..daca9264 100644
--- a/tensorflow_serving/model_servers/BUILD
+++ b/tensorflow_serving/model_servers/BUILD
@@ -407,6 +407,7 @@ cc_binary(
         "//tensorflow_serving:internal",
     ],
     deps = [
+       "@openvino_tensorflow//:openvino_tensorflow",
         ":tensorflow_model_server_main_lib",
     ],
 )
