diff --git a/examples/mnist_mlp.py b/examples/mnist_mlp.py
index cba180ff..750d1d84 100644
--- a/examples/mnist_mlp.py
+++ b/examples/mnist_mlp.py
@@ -12,16 +12,29 @@ from keras.datasets import mnist
 from keras.models import Sequential
 from keras.layers import Dense, Dropout
 from keras.optimizers import RMSprop
+import ngraph_bridge
+import tensorflow as tf
+
+from keras.backend.tensorflow_backend import set_session
+config = tf.compat.v1.ConfigProto()
+config = ngraph_bridge.update_config(config)
+sess = tf.compat.v1.Session(config=config)
+tf.compat.v1.keras.backend.set_session(sess)
 
 batch_size = 128
 num_classes = 10
-epochs = 20
+epochs = 1
 
 # the data, split between train and test sets
 (x_train, y_train), (x_test, y_test) = mnist.load_data()
 
 x_train = x_train.reshape(60000, 784)
 x_test = x_test.reshape(10000, 784)
+max_samples = 1024
+x_train = x_train[:max_samples, :]
+x_test = x_test[:max_samples, :]
+y_train = y_train[:max_samples]
+y_test = y_test[:max_samples]
 x_train = x_train.astype('float32')
 x_test = x_test.astype('float32')
 x_train /= 255
