diff --git a/tensorflow/python/framework/test_util.py b/tensorflow/python/framework/test_util.py
index bf00fa6..23e0265 100644
--- a/tensorflow/python/framework/test_util.py
+++ b/tensorflow/python/framework/test_util.py
@@ -29,6 +29,7 @@ import threading
 
 import numpy as np
 import six
+from ctypes import *
 
 _portpicker_import_error = None
 try:
@@ -757,6 +758,9 @@ class TensorFlowTestCase(googletest.TestCase):
     self._threads = []
     self._tempdir = None
     self._cached_session = None
+    cdll.LoadLibrary(
+    'libngraph_device.so'
+    )
 
   def setUp(self):
     self._ClearCachedSession()
@@ -965,6 +969,7 @@ class TensorFlowTestCase(googletest.TestCase):
       if config is None:
         config = config_pb2.ConfigProto()
         config.allow_soft_placement = not force_gpu
+        config.log_device_placement=True
         config.gpu_options.per_process_gpu_memory_fraction = 0.3
       elif force_gpu and config.allow_soft_placement:
         config = config_pb2.ConfigProto().CopyFrom(config)
@@ -993,7 +998,7 @@ class TensorFlowTestCase(googletest.TestCase):
         elif use_gpu:
           yield sess
         else:
-          with sess.graph.device("/cpu:0"):
+          with sess.graph.device("/device:NGRAPH:0"):
             yield sess
     else:
       with session.Session(graph=graph, config=prepare_config(config)) as sess:
@@ -1008,7 +1013,7 @@ class TensorFlowTestCase(googletest.TestCase):
         elif use_gpu:
           yield sess
         else:
-          with sess.graph.device("/cpu:0"):
+          with sess.graph.device("/device:NGRAPH:0"):
             yield sess
 
   # pylint: enable=g-doc-return-or-yield
