diff --git a/tensorflow/tools/ci_build/builds/pip_new.sh b/tensorflow/tools/ci_build/builds/pip_new.sh
index 27208591d50..d89fa862569 100755
--- a/tensorflow/tools/ci_build/builds/pip_new.sh
+++ b/tensorflow/tools/ci_build/builds/pip_new.sh
@@ -315,6 +315,8 @@ bazel build \
   --action_env=PYTHON_BIN_PATH=${PYTHON_BIN_PATH} \
   ${TF_BUILD_FLAGS} \
   ${PIP_BUILD_TARGET} \
+  --local_ram_resources=12000 --jobs=8 --local_cpu_resources=8 \
+  --config=noaws --config=nohdfs --config=nonccl --cxxopt="-D_GLIBCXX_USE_CXX11_ABI=1" \
   || die "Error: Bazel build failed for target: '${PIP_BUILD_TARGET}'"
 
 ###########################################################################
@@ -696,7 +698,7 @@ if [[ "$BUILD_BOTH_GPU_PACKAGES" -eq "1" ]] || [[ "$BUILD_BOTH_CPU_PACKAGES" -eq
 fi
 
 # Run tests (if any is specified).
-run_all_tests
+#run_all_tests
 
 
 if [[ ${OS_TYPE} == "ubuntu" ]] && \

