diff --git a/tensorflow_scripts/DRAW/draw_inf.py b/tensorflow_scripts/DRAW/draw_inf.py
index ce88f35b..00a8d5a2 100644
--- a/tensorflow_scripts/DRAW/draw_inf.py
+++ b/tensorflow_scripts/DRAW/draw_inf.py
@@ -15,6 +15,7 @@ import numpy as np
 import os
 import time
 import ngraph_bridge
+from tensorflow.core.protobuf import rewriter_config_pb2
 
 tf.flags.DEFINE_string("dl", "./", "")
 tf.flags.DEFINE_string("cp", "./drawmodel.ckpt", "checkpoint file")
@@ -228,6 +229,10 @@ Lzs=[0]*train_iters
 config = tf.ConfigProto(
     inter_op_parallelism_threads=FLAGS.num_inter_threads, 
     intra_op_parallelism_threads=FLAGS.num_intra_threads)
+rewrite_options = rewriter_config_pb2.RewriterConfig(
+    meta_optimizer_iterations=rewriter_config_pb2.RewriterConfig.ONE,
+    custom_optimizers=[rewriter_config_pb2.RewriterConfig.CustomGraphOptimizer(name="ngraph-optimizer")])
+config.graph_options.rewrite_options.MergeFrom(rewrite_options)
 sess=tf.Session(config = config)
 
 saver = tf.train.Saver() # saves variables learned during training
