diff --git a/tools/build_utils.py b/tools/build_utils.py
index 60e74c35..c0adc8ee 100755
--- a/tools/build_utils.py
+++ b/tools/build_utils.py
@@ -853,17 +853,12 @@ def build_openvino(build_dir, openvino_src_dir, cxx_abi, target_arch,
     else:
         openvino_cmake_flags.extend([
             "-DNGRAPH_COMPONENT_PREFIX=deployment_tools/ngraph/",
-            "-DCMAKE_INSTALL_PREFIX=" + install_location
+            "-DCMAKE_INSTALL_PREFIX=" + install_location, "-DTREAT_WARNING_AS_ERROR=OFF"
         ])
 
-    if platform.system() == 'Linux':
-        openvino_cmake_flags.extend([
-            "-DCMAKE_CXX_FLAGS=-D_GLIBCXX_USE_CXX11_ABI=" + cxx_abi + " -march="
-            + target_arch + atom_flags
-        ])
-    else:
-        openvino_cmake_flags.extend(
-            ["-DCMAKE_CXX_FLAGS=-D_GLIBCXX_USE_CXX11_ABI=" + cxx_abi])
+
+    openvino_cmake_flags.extend(
+        ["-DCMAKE_CXX_FLAGS=-D_GLIBCXX_USE_CXX11_ABI=" + cxx_abi])
 
     if debug_enabled:
         openvino_cmake_flags.extend(["-DCMAKE_BUILD_TYPE=Debug"])

