diff --git a/official/resnet/cifar10_main.py b/official/resnet/cifar10_main.py
index e5e5008f..edfe77dc 100644
--- a/official/resnet/cifar10_main.py
+++ b/official/resnet/cifar10_main.py
@@ -28,6 +28,7 @@ from official.resnet import resnet_model
 from official.resnet import resnet_run_loop
 from official.utils.flags import core as flags_core
 from official.utils.logs import logger
+import ngraph_bridge
 
 HEIGHT = 32
 WIDTH = 32
@@ -237,10 +238,11 @@ def define_cifar_flags():
   flags_core.set_defaults(data_dir='/tmp/cifar10_data/cifar-10-batches-bin',
                           model_dir='/tmp/cifar10_model',
                           resnet_size='56',
-                          train_epochs=182,
+                          train_epochs=1,
                           epochs_between_evals=10,
                           batch_size=128,
-                          image_bytes_as_serving_input=False)
+                          image_bytes_as_serving_input=False,
+                          max_train_steps=100)
 
 
 def run_cifar(flags_obj):
