diff --git a/tensorflow/python/framework/test_util.py b/tensorflow/python/framework/test_util.py
index 544190e23d..19e621478e 100644
--- a/tensorflow/python/framework/test_util.py
+++ b/tensorflow/python/framework/test_util.py
@@ -2737,7 +2737,9 @@ class TensorFlowTestCase(googletest.TestCase):
           rewriter_config_pb2.RewriterConfig.OFF)
       config.graph_options.rewrite_options.pin_to_host_optimization = (
           rewriter_config_pb2.RewriterConfig.OFF)
-      return config
+      import ngraph_bridge
+      config_new = ngraph_bridge.update_config(config)
+      return config_new
 
     return ErrorLoggingSession(graph=graph, config=prepare_config(config))
 
