diff --git a/tensorflow_core/python/framework/test_util.py b/tensorflow_core/python/framework/test_util.py
index 8857e767ed..3dcd7d17b2 100644
--- a/tensorflow_core/python/framework/test_util.py
+++ b/tensorflow_core/python/framework/test_util.py
@@ -1767,6 +1767,7 @@ class TensorFlowTestCase(googletest.TestCase):
     self._threads = []
     self._tempdir = None
     self._cached_session = None
+    import ngraph_bridge
 
   def setUp(self):
     self._ClearCachedSession()
