diff --git a/tensorflow/python/framework/test_util.py b/tensorflow/python/framework/test_util.py
index 544190e23d..9b652bd49e 100644
--- a/tensorflow/python/framework/test_util.py
+++ b/tensorflow/python/framework/test_util.py
@@ -1662,6 +1662,9 @@ class TensorFlowTestCase(googletest.TestCase):
     self._threads = []
     self._tempdir = None
     self._cached_session = None
+    import ngraph_bridge
+
+    ngraph_bridge.set_disabled_ops('NoOp')
 
   def setUp(self):
     self._ClearCachedSession()