--- test_util.py	2020-12-22 14:33:06.758700689 -0800
+++ updated_test_util.py	2021-01-04 10:31:39.382524408 -0800
@@ -3121,7 +3121,9 @@
           rewriter_config_pb2.RewriterConfig.OFF)
       config.graph_options.rewrite_options.pin_to_host_optimization = (
           rewriter_config_pb2.RewriterConfig.OFF)
-      return config
+      import ngraph_bridge
+      config_new = ngraph_bridge.update_config(config)
+      return config_new
 
     return ErrorLoggingSession(graph=graph, config=prepare_config(config))
 
