diff --git a/README.md b/README.md
index ae78b340..14f9bc1c 100644
--- a/README.md
+++ b/README.md
@@ -1 +1,3 @@
 # ngraph-models
+
+# Dummy change because the test framework expects a patch file. Generally that is true for other models, but ngraph-models already has ngraph_bridge patched in
diff --git a/tensorflow_scripts/run-all-models.sh b/tensorflow_scripts/run-all-models.sh
index 03b63406..58b07756 100755
--- a/tensorflow_scripts/run-all-models.sh
+++ b/tensorflow_scripts/run-all-models.sh
@@ -21,34 +21,34 @@ set -o nounset
 declare -a TEST_MODELS=(
     "run-a3c.sh;./tools/calc-tf-a3c-results.py"
     "run-dcgan.sh;./tools/calc-dcgan-results.py"
-    "run-densenet_40_12.sh"
+    #"run-densenet_40_12.sh"
     "run-densenet.sh;./tools/calc-tf-densenet-results.py"
-    "run-draw.sh;./tools/calc-draw-results.py"
-    "run-fasterRCNN.sh;./tools/calc-tf-fasterRCNN-results.py"
-    "run-gnmt-de-en.sh"
-    "run-gnmt-en-vi.sh"
-    #"run-gnmt-en-vi-training.sh" # Takes a long time. Need to be run with a different script
-    "run-inception-resnet-v2.sh;./tools/calc-tf-inceptionresnetv2-results.py"
-    "run-inception-v3.sh;./tools/calc-tf-cnn-resnet50_v2_results.py"
-    "run-inception-v4.sh;./tools/calc-tf-cnn-resnet50_v2_results.py"
-    "run-mask-rcnn.sh"
-    "run-mobilenet-v2.sh;./tools/calc-tf-cnn-resnet50_v2_results.py"
-    "run-mobilenet-v1.sh;./tools/calc-MobileNet_v1-results.py"
-    "run-ncf.sh;./tools/calc-ncf-results.py"
-    "run-resnet50-v1.sh;./tools/calc-tf-cnn-resnet50_v2_results.py"
-    "run-resnet50-v2.sh;./tools/calc-tf-cnn-resnet50_v2_results.py"
+    #"run-draw.sh;./tools/calc-draw-results.py"
+    #"run-fasterRCNN.sh;./tools/calc-tf-fasterRCNN-results.py"
+    #"run-gnmt-de-en.sh"
+    #"run-gnmt-en-vi.sh"
+    ##"run-gnmt-en-vi-training.sh" # Takes a long time. Need to be run with a different script
+    #"run-inception-resnet-v2.sh;./tools/calc-tf-inceptionresnetv2-results.py"
+    #"run-inception-v3.sh;./tools/calc-tf-cnn-resnet50_v2_results.py"
+    #"run-inception-v4.sh;./tools/calc-tf-cnn-resnet50_v2_results.py"
+    #"run-mask-rcnn.sh"
+    #"run-mobilenet-v2.sh;./tools/calc-tf-cnn-resnet50_v2_results.py"
+    #"run-mobilenet-v1.sh;./tools/calc-MobileNet_v1-results.py"
+    #"run-ncf.sh;./tools/calc-ncf-results.py"
+    #"run-resnet50-v1.sh;./tools/calc-tf-cnn-resnet50_v2_results.py"
+    #"run-resnet50-v2.sh;./tools/calc-tf-cnn-resnet50_v2_results.py"
     #"run-rfcn-non-frozen.sh" # Error: JIRA filed
-    "run-rfcn.sh;./tools/calc-tf-rfcn-results.py"
-    "run-squeezenet_v1_1.sh;./tools/calc-tf-squeezenet-results.py"
-    #"run-ssd-mobilenet-v1-non-frozen.sh" # Hangs at the end
-    "run-ssd-mobilenet-v1.sh;./tools/calc-tf-ssd-results.py"
-    "run-ssd-vgg16.sh;./tools/calc-ssd-vgg16-results.py"
-    "run-transformer-LT.sh" 
-    "run-unet.sh;./tools/calc-tf-unet-results.py"
-    "run-vgg16.sh;./tools/calc-vgg16-results.py"
-    "run-wavenet.sh;./tools/calc-wavenet-results.py"
-    "run-wide-and-deep.sh;./tools/calc-wide-and-deep-results.py"
-    "run-yolo-v2.sh;./tools/calc-yolov2-results.py"
+    #"run-rfcn.sh;./tools/calc-tf-rfcn-results.py"
+    #"run-squeezenet_v1_1.sh;./tools/calc-tf-squeezenet-results.py"
+    ##"run-ssd-mobilenet-v1-non-frozen.sh" # Hangs at the end
+    #"run-ssd-mobilenet-v1.sh;./tools/calc-tf-ssd-results.py"
+    #"run-ssd-vgg16.sh;./tools/calc-ssd-vgg16-results.py"
+    #"run-transformer-LT.sh"
+    #"run-unet.sh;./tools/calc-tf-unet-results.py"
+    #"run-vgg16.sh;./tools/calc-vgg16-results.py"
+    #"run-wavenet.sh;./tools/calc-wavenet-results.py"
+    #"run-wide-and-deep.sh;./tools/calc-wide-and-deep-results.py"
+    #"run-yolo-v2.sh;./tools/calc-yolov2-results.py"
 )
 
 rm -f failed.txt
